src/backend/distributed/commands/extension.c

```diff
@@ -340,7 +338,7 @@ FilterDistributedExtensions(List *extensionObjectList)
 			continue;
 		}
 
-		extensionNameList = lappend(extensionNameList, makeString(extensionName));
+		extensionNameList = lappend(extensionNameList, objectName);
```

Comment:
Made the decision to copy the `Value*` since if we're wanting to create a new `Value*` we should also be calling `pstrdup` on `extensionName`. This choice is repeated in this commit